### PR TITLE
Initial implementation of --locked

### LINKED ISSUE #5: Implement `--locked` argument for install command
This is a critical bit for ensuring that we always check in the most up-to-date lockfile.

### PR DESCRIPTION
**Pull request by u-train**
Closes #5 with an implementation `--locked` for install.

### CODE REVIEW

File: src/commands/install.rs [hunk 1]
```diff
@@ -43,29 +51,64 @@ impl InstallSubcommand {
...
+                    "{}{} Suggestion{}{} try running wally update",
+                    SetAttribute(Attribute::Bold),
+                    SetForegroundColor(Color::DarkGreen),
+                    SetForegroundColor(Color::Reset),
+                    SetAttribute(Attribute::Reset)
+                )?;
+
+                // Should be safe since we're only getting valid utf-8.
+                anyhow::bail!(String::from_utf8(error_output).unwrap());
             }
```
Comment (Dekkonot) [inline src/commands/install.rs:97]:
```suggestion
                anyhow::bail!(String::from_utf8(error_output)
                    .expect("output from render_update_difference should always be utf-8"));
```
`expect` is better if you expect it to never fail. `unwrap` is more of a "I don't care if it fails".

File: src/commands/install.rs [hunk 2]
```diff
@@ -43,29 +51,64 @@ impl InstallSubcommand {
...
+            ));
+
+            let latest_graph = resolve(&manifest, &BTreeSet::new(), &package_sources)?;
+
+            if try_to_use != latest_graph.activated {
+                progress.finish_and_clear();
+
+                let old_dependencies = &try_to_use;
+
+                let changes = generate_depedency_changes(old_dependencies, &latest_graph.activated);
```
Comment (Dekkonot) [inline src/commands/install.rs:76]:
```suggestion
                let changes = generate_dependency_changes(old_dependencies, &latest_graph.activated);
```
This was spelled wrong before too, I just think we can fix it while we're moving it around.

File: src/commands/update.rs [hunk 3]
```diff
@@ -16,6 +15,8 @@ use crossterm::style::{Attribute, Color, SetAttribute, SetForegroundColor};
 use indicatif::{ProgressBar, ProgressStyle};
 use structopt::StructOpt;
 
+use super::utils::{generate_depedency_changes, render_update_difference};
```
Comment (Dekkonot) [inline src/commands/update.rs:18]:
```suggestion
use super::utils::{generate_dependency_changes, render_update_difference};
```

File: src/commands/utils.rs [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+use std::{collections::BTreeSet, io::Write};
+
+pub(crate) enum DependencyChange {
+    Added(PackageId),
+    Removed(PackageId),
+    Updated { from: PackageId, to: PackageId },
+    Downgrade { from: PackageId, to: PackageId },
+}
+
+pub(crate) fn generate_depedency_changes(
```
Comment (Dekkonot) [inline src/commands/utils.rs:12]:
```suggestion
pub(crate) fn generate_dependency_changes(
```

File: src/commands/install.rs [hunk 5]
```diff
@@ -43,29 +51,64 @@ impl InstallSubcommand {
...
-        }
 
-        let progress = ProgressBar::new(0)
-            .with_style(
-                ProgressStyle::with_template("{spinner:.cyan}{wide_msg}")?.tick_chars("⠁⠈⠐⠠⠄⠂ "),
-            )
-            .with_message(format!(
-                "{} Resolving {}packages...",
+            progress.println(format!(
+                "{} Verified {}lockfile is up-to-date...{}",
```
Comment (Dekkonot) [inline src/commands/install.rs:100]:
```suggestion
                "{}   Verified {}lockfile is up-to-date...{}",
```
This is silly but it looks better with more spaces. This aligns it with the "Resolved dependencies" message that happens later in the install process.

**Review by Dekkonot — CHANGES_REQUESTED** [review]

File: src/commands/install.rs (hunk 1 above)
Comment (Dekkonot) [reply src/commands/install.rs:97]: Oh good, VSCode messed up the alignment on this one.

Hopefully you know what I meant.

Comment (u-train) [reply src/commands/install.rs:97]: I did lol

**Review by Dekkonot — COMMENTED** [review]
This needs an entry in the changelog but otherwise it looks good to me.

**Review by Dekkonot — APPROVED** [review]
lgtm

**Pull request merged**